File: Iter.Api/Iter.Core/Search Responses/ArrangementSearchResponse.cs
namespace Iter.Core
{
    public class ArrangementSearchResponse
    {
        public Guid Id { get; set; }

        public string Name { get; set; }

        public string AgencyName { get; set; }

        public decimal AgencyRating { get; set; }

        public DateTime StartDate { get; set; }

        public DateTime? EndDate { get; set; }

        public int? ArrangementStatusId { get; set; }

        public string ArrangementStatusName { get; set; }

        public ImageResponse? MainImage { get; set; }

        public bool IsReserved { get; set; }
    }
}

File: Iter.Api/ui/Iter.Desktop/lib/models/arrangement_search_response.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:ui/models/image_model.dart';

part 'arrangement_search_response.g.dart';

@JsonSerializable()
class ArrangementSearchResponse {
  final String id;
  final String name;
  final String agencyName;
  final double agencyRating;
  final DateTime startDate;
  DateTime? endDate;
  int? arrangementStatusId;
  String? arrangementStatusName;
  ImageModel? mainImage;
  bool? isReserved;
  double? rating;

  ArrangementSearchResponse({
    required this.id,
    required this.name,
    required this.agencyName,
    required this.agencyRating,
    required this.startDate,
    this.endDate,
    this.arrangementStatusId,
    this.arrangementStatusName,
    this.mainImage,
    this.isReserved,
    this.rating
  });

  factory ArrangementSearchResponse.fromJson(Map<String, dynamic> json) =>
      _$ArrangementSearchResponseFromJson(json);

  Map<String, dynamic> toJson() => _$ArrangementSearchResponseToJson(this);
}
